osbs/build/build_request.py

```diff
@@ -1186,7 +1195,22 @@ def render(self, validate=True):
         if koji_task_id is not None:
             self.set_label('koji-task-id', str(koji_task_id))
 
-        self.template['spec']['nodeSelector'] = self.platform_node_selector
+        # for worker builds set nodeselectors
+        if self.spec.platforms.value is None:
+
+            # auto or explicit build selector
+            if self.is_auto:
+                self.template['spec']['nodeSelector'] = self.auto_build_node_selector
+            else:
+                self.template['spec']['nodeSelector'] = self.explicit_build_node_selector
+
+            # platform nodeselector
+            if self.platform_node_selector:
+                self.template['spec']['nodeSelector'].update(self.platform_node_selector)
+
+            # scratch build nodeselector
+            if self.scratch:
```

Comment:
that would add only scratch and not 'scratch & explicit'

yeah I know that by definition you can't have that